crates/katana/rpc/src/api/starknet.rs

```diff
@@ -16,73 +16,110 @@ use katana_rpc_types::transaction::{
-    ClassAlreadyDeclared = 51,
+    ClassAlreadyDeclared,
     #[error("Invalid transaction nonce")]
-    InvalidTransactionNonce = 52,
+    InvalidTransactionNonce,
     #[error("Max fee is smaller than the minimal transaction cost (validation plus fee transfer)")]
-    InsufficientMaxFee = 53,
+    InsufficientMaxFee,
     #[error("Account balance is smaller than the transaction's max_fee")]
-    InsufficientAccountBalance = 54,
+    InsufficientAccountBalance,
     #[error("Account validation failed")]
-    ValidationFailure = 55,
+    ValidationFailure,
     #[error("Compilation failed")]
-    CompilationFailed = 56,
+    CompilationFailed,
     #[error("Contract class size is too large")]
-    ContractClassSizeIsTooLarge = 57,
+    ContractClassSizeIsTooLarge,
     #[error("Sender address in not an account contract")]
-    NonAccount = 58,
+    NonAccount,
     #[error("A transaction with the same hash already exists in the mempool")]
-    DuplicateTransaction = 59,
+    DuplicateTransaction,
     #[error("The compiled class hash did not match the one supplied in the transaction")]
-    CompiledClassHashMismatch = 60,
+    CompiledClassHashMismatch,
     #[error("The transaction version is not supported")]
-    UnsupportedTransactionVersion = 61,
+    UnsupportedTransactionVersion,
     #[error("The contract class version is not supported")]
-    UnsupportedContractClassVersion = 62,
+    UnsupportedContractClassVersion,
     #[error("An unexpected error occured")]
-    UnexpectedError = 63,
+    UnexpectedError,
     #[error("Too many storage keys requested")]
-    ProofLimitExceeded = 10000,
+    ProofLimitExceeded,
     #[error("Too many keys provided in a filter")]
-    TooManyKeysInFilter = 34,
+    TooManyKeysInFilter,
     #[error("Failed to fetch pending transactions")]
-    FailedToFetchPendingTransactions = 38,
+    FailedToFetchPendingTransactions,
 }
 
 impl From<StarknetApiError> for Error {
     fn from(err: StarknetApiError) -> Self {
-        Error::Call(CallError::Custom(ErrorObject::owned(err as i32, err.to_string(), None::<()>)))
+        let message = err.to_string();
+
+        let (code, data) = match err {
+            StarknetApiError::FailedToReceiveTxn => (1, None),
+            StarknetApiError::ContractNotFound => (20, None),
+            StarknetApiError::InvalidMessageSelector => (21, None),
+            StarknetApiError::InvalidCallData => (22, None),
+            StarknetApiError::BlockNotFound => (24, None),
+            StarknetApiError::TxnHashNotFound => (29, None),
+            StarknetApiError::InvalidTxnIndex => (27, None),
+            StarknetApiError::ClassHashNotFound => (28, None),
+            StarknetApiError::PageSizeTooBig => (31, None),
+            StarknetApiError::NoBlocks => (32, None),
+            StarknetApiError::InvalidContinuationToken => (33, None),
+            StarknetApiError::ContractError { revert_error } => {
+                (40, Some(ContractErrorData { revert_error }))
```

Comment:
then, we can start pattern match for this variant only

Reply:
Thank you for this, fixed in ba0fdd465a7b3c10f9ca855599d208d6cc50ba90.